### ISSUE TITLE
Leading zeroes get stripped from KvK and KvK-branch number fields

### ISSUE THREAD

**Issue by hlmr-erik** [open-formulieren/open-forms#1903]
### Product versie / Product version

1.1.5

### Omschrijf het probleem / Describe the bug

When creating fields with the validation plug-ins for KvK number and KvK-branch number fields, filling these in with a number that has trailing zeroes, makes these get stripped away - and therefore the validation fails as the numbers aren't valid anymore.

KvK branch numbers very commonly have trailing zeroes, they do exist in regular KvK numbers as well but are a lot less common. This is probably applicable to KvK RSIN numbers as well, yet I haven't tested this.

![chrome_2022-08-22_15 29 17](https://user-images.githubusercontent.com/111565013/185933618-f02d63d4-749c-434a-a415-b9330c7aa6c9.gif)

### Stappen om te reproduceren / Steps to reproduce

1. Go to https://haarlemmermeer.test.open-formulieren.nl/ttttest/stap/test2
2. Fill '000041571800' in the field for 'KvK Vestigingsnummer'

### Verwacht gedrag / Expected behavior

Fields should validate without stripping leading zeroes 

### Device

Desktop

### OS

Windows

### Browser

Chrome, Firefox

**Comment by sergei-maertens** [comment]
Can you confirm that you are NOT using a `number` type component, but a regular textfield for this?

**Comment by hlmr-erik** [comment]
> Can you confirm that you are NOT using a `number` type component, but a regular textfield for this?

I was indeed using a `number` type compenent as, well -it is a number-. Changing it to a textfield solved it indeed, so this behaviour could be fine if the documentation makes it clear (and maybe disallow to set these validation plugins on `number` components?).

**Comment by sergei-maertens** [comment]
yeah, it's tricky, you cannot use *actual* number-based datatypes for what are essentially *numeric strings* - this is a common (programmer) mistake :smile: 

I'll leave the issue open as a documentation issue, this can probably be mentioned somewhere in the manual.

> In case it is intended behaviour, maybe disallow to set these validation plugins on number components?

Validators are standalone and completely decoupled from the context that they're used in - so this is at the moment not easily possible. I think we have an existing issue for that somewhere, at least I remember discussing it with colleagues.

**Comment by StefSchots1993** [comment]
@joeribekker @LaurensBurger  - N.a.v. refinement met Haarlemmermeer even de vraag, kunnen jullie ervoor zorgen dat dit wordt opgenomen in de documentatie op https://open-forms.readthedocs.io/en/stable/index.html
